In [1]:
import tweepy
import time

In [3]:
# Claves del consumidor y tokens de acceso, utilizados para OAuth
consumer_key = 'd1txrogo10pOoDbgah9WDmyGx'
consumer_secret = 'oB5Xrnfb7wrMJLip0ZLNLO3blRgNfGuA4ZBNyzoY7KIy3v5Qya'

consumer_Owner = '_Huguez'
consumer_Owner_ID = '714880949876461568'

access_token = '714880949876461568-Jy0zkv22pLrJdxoSnWn8s31TdfRfyUI'
access_token_secret = 'AhkzMdrfk9tBawrpUYdh0vXoRRERI4BFK40ak5KbnZ3ua'

# Proceso de OAuth, utilizando las claves y los tokens
auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_token_secret)

# autenticación, usando una interfaz de la api de twitter
api = tweepy.API(auth, wait_on_rate_limit=True )


In [4]:

def llenarTweetts( twittero, listTwitteros, nivel ):
    
    def limit_handled(cursor):
        while True:
            try:
                yield cursor.next()
            except tweepy.RateLimitError:
                time.sleep( 15*60 )
                continue
            except StopIteration:
                break

    
    # con este for recorro todos los followers de mi tweetter
    for follower in api.followers( twittero.screen_name ):

        # las cuantas que contiene menos de 300 seguinores son seleccionadas 
        if follower.friends_count < 300:
            # si le twittero no esta en el diccionario
            # continuamos con la operacion
            if not follower.protected and follower.screen_name not in listTwitteros:
                
                # incluyo los nombres como llaves en el diccionario donde
                # los valores sera una lista de todos los twitteros que siguen
                listTwitteros[ follower.screen_name ] = list()

                # lleno la lista con los twitteros
                listTwitteros[ follower.screen_name ] = [ tweet.text for tweet in api.user_timeline(follower.screen_name,count = 200 ) ]        
            
            # llamamos a la funcion ora ves pero con el seguidor    
            if nivel > 0:
                if not follower.protected:
                    llenarTweetts( follower, listTwitteros, nivel-1 )


In [7]:

bdTwitteros = {}

api = tweepy.API(auth, wait_on_rate_limit=True )

llenarTweetts( api.me(), bdTwitteros, 3 )


k = 0
for i in bdTwitteros:
    k += len( bdTwitteros[i] )
    #print( k )
    
print( k, len(bdTwitteros) )



296560 3143


Como podemos ver arriba se lograron obtener 296560 tweets de 3143 usuarios de twitter exitosamente, con estos empesaremos a trabajar

In [14]:
databases = bdTwitteros.copy()
    

In [27]:
import csv


#dicc = { 'first_name': [ 'Baked', 'Lovely', 'Wonderful' ], 'last_name': [ 'Beans','Spam', 'Spam2' ] }

with open('db.csv','w') as csvfile:
    fieldnames = ['twittero', 'tweett']
    
    writer = csv.DictWriter(csvfile, fieldnames=fieldnames)

    writer.writeheader()
    
    for key in bdTwitteros:
        for text in bdTwitteros[key]:
            
            #print("{}{}".format( key, text )  )
            writer.writerow( {'twittero': key, 'tweett': text })
            
            